CHANGELOG.md

```diff
@@ -3,7 +3,146 @@
+      }
+  
+      return ( /* Some UI to edit the date */ )
+    }
+  
+    function MyCustomField(props) {
+      const fieldResponse = useDateField<Dayjs, false, typeof textFieldProps>({
+        ...props,
+  +     // If you only support one DOM structure, we advise you to hardcode it here to avoid unwanted switches in your application
+  +     enableAccessibleFieldDOMStructure: false,
+      });
+  
+      return <MyCustomTextField ref={ref} {...fieldResponse} />;
+    }
+  
+    function App() {
+      return <DatePicker slots={{ field: MyCustomField }} />;
+    }
+  ```
+
+- The following internal types were exported by mistake and have been removed from the public API:
+
+  - `UseDateFieldDefaultizedProps`
+  - `UseTimeFieldDefaultizedProps`
+  - `UseDateTimeFieldDefaultizedProps`
+  - `UseSingleInputDateRangeFieldComponentProps`
+  - `UseSingleInputTimeRangeFieldComponentProps`
+  - `UseSingleInputDateTimeRangeFieldComponentProps`
+
+
+#### `@mui/x-date-pickers@7.0.0-beta.4`
+
+- [fields] Use the `PickersTextField` component in the fields (#10649) @flaviendelangle
+- [pickers] Fix styling props propagation to `DateTimePickerTabs` (#12096) @LukasTy
+
+#### `@mui/x-date-pickers-pro@7.0.0-beta.4` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@7.0.0-beta.4`.
+
+### Charts / `@mui/x-charts@7.0.0-beta.4`
+
+#### Breaking changes
+
+Those components are not longer exported by `@mui/x-charts`:
+
+- `CartesianContextProvider`
+- `DrawingProvider`
+
+#### `@mui/x-charts@7.0.0-beta.4`
+
+- [charts] Don't display text if no value provided (#12127) @alexfauquette
+- [charts] Remove export of context providers (#12123) @oliviertassinari
+
+### Tree View / `@mui/x-tree-view@7.0.0-beta.4`
+
+- [TreeView] Stop using custom `findIndex` to support IE11 (#12129) @flaviendelangle
+
+### Docs
+
+- [docs] Add recipe for hiding separator on non-resizable columns (#12134) @michelengelen
+- [docs] Clarify Pickers `Component composition` section (#12097) @LukasTy
+- [docs] Fix "Licensing" page link (#12156) @LukasTy
+- [docs] Fix error raised by Grammarly on the page @oliviertassinari
+- [docs] Improve performance on charts entry point @oliviertassinari
+- [docs] Link to react-transition group with https @oliviertassinari
+- [docs] Move Heat map to pro (#12047) @alexfauquette
+- [docs] Reduce number of Vale errors @oliviertassinari
+- [docs] Remove default value set to undefined (#12128) @alexfauquette
+
+### Core
+
+- [core] Fix docs link check (#12135) @LukasTy
+- [core] Fix missing context display names (#12124) @oliviertassinari
+- [core] Fix shortcuts when Caps Lock enabled (#12121) @oliviertassinari
+- [core] Remove IE 11 compat logic (#12119) @oliviertassinari
+- [core] Simplify key utils (#12120) @oliviertassinari
+- [core] Use the @mui/internal-scripts package (#12142) @michaldudak
+- [all components] Remove legacy IE 11 bundle (#12151) @flaviendelangle
```

Comment:
@MBilalShafi What are we doing with this one? 🤔 
It impacts all the packages and, as mentioned, is a breaking change.
WDYT about listing it in highlights and also on every package?

Reply:
I guess adding a highlight would be enough. Wdyt?
Also, it feels if it is not _technically_ a breaking change since the change is not done to the API of any component, would you put it in breaking changes too?

Reply:
Well, it's not a BC to everyone, but still a BC in the sense that after this we no longer support legacy browsers like IE11. 🤔 
IMHO, it would be enough to list it in the highlights only.
But with a clear description.
Like:
```suggestion
- 🕥 The support for IE11 has been removed (#12151) @flaviendelangle
  We no longer provide a `legacy` bundle that used to support old browsers like IE11.
```

WDYT @flaviendelangle

Reply:
> Also, it feels if it is not technically a breaking change since the change is not done to the API of any component

IMHO this is a breaking change
People can have an app that works before and that no longer works for some of their users.

Strictly speaking I think this should be listed in each set of components.
But it's a limitation of our current README structure, anything related to all components is super redundant if we want to apply it correctly to all sets of components.

Given the very low usage of IE11 and the fact that I think we were already not compatible (we had usages of `arr.findIndex` in the app, which IE11 does not support and in one place a small polyfill), I'd be fine only adding it in the highlights.

Reply:
I suggested a top-level breaking change for this change, let me know what you think about it.
We could add other BCs that apply to all components in that section (for future)